packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/transfer/database/database_io.dart

```diff
@@ -51,16 +52,29 @@ class TransferDatabase extends _$TransferDatabase {
               dateTime.toIso8601String(),
             ),
           ))
+        .map(
+          (e) => data.TransferRecord(
+            objectKey: e.objectKey,
+            uploadId: e.uploadId,
+            createdAt: e.createdAt,
+          ),
+        )
         .get();
   }
 
-  /// Inserts [entry], a [TransferRecordsCompanion], to the database.
-  Future<int> insertTransferRecord(TransferRecordsCompanion entry) {
-    return into(transferRecords).insert(entry);
+  @override
+  Future<String> insertTransferRecord(data.TransferRecord record) {
+    final entry = TransferRecordsCompanion.insert(
+      uploadId: record.uploadId,
+      objectKey: record.objectKey,
+      createdAt: record.createdAt,
+    );
+    return into(transferRecords)
+        .insert(entry)
+        .then((value) => value.toString());
```

Comment:
Prefer async/await when in an async context. Control flow is more clear.

```suggestion
    final value = await into(transferRecords).insert(entry);
    return value.toString();
```

Reply:
[effective dart](https://dart.dev/guides/language/effective-dart/usage#avoid-using-completer-directly) considers both async/await and future.then as good style patterns. I might be missing other style guides that we should follow so please lmk.

Reply:
Both are fine, but async/await is preferred: https://dart.dev/guides/language/effective-dart/usage#prefer-asyncawait-over-using-raw-futures

Reply:
I would say only use `.then` in situations where you explicitly want to avoid async/await, like when you want to fire-and-forget a future. But 95% of the time, async/await is easier and will make the code more readable.

Reply:
thanks for sharing the ref and explanation. will update the code